OSDOCS-8904: Private Cluster on ROSA with HCP docs (#70451)
openshift/openshift-docs: modules/rosa-hcp-aws-privatelink-create-cluster.adoc

// Module included in the following assemblies:
//
// * rosa_hcp/rosa-hcp-aws-privatelink-creating-cluster.adoc
:_mod-docs-content-type: PROCEDURE
[id="rosa-hcp-aws-privatelink-create-cluster_{context}"]
= Creating an AWS PrivateLink cluster

You can create a private cluster with multiple availability zones (Multi-AZ) on {hcp-title} using the ROSA command line interface (CLI), `rosa`.

[NOTE]
====
AWS PrivateLink is supported on preexisting VPCs only.
====

.Prerequisites

* You have available AWS service quotas.
* You have enabled the ROSA service in the AWS Console.
* You have installed and configured the latest version of the ROSA CLI on your installation host.

.Procedure

Creating a cluster with {hcp} can take around 10 minutes.

> . Create a VPC with at least one private subnet. Ensure that your machine's classless inter-domain routing (CIDR) matches your virtual private cloud's CIDR. See https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-vpc.html#installation-custom-aws-vpc-requirements_installing-aws-vpc[Requirements for using your own VPC] and link:https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-vpc.html#installation-custom-aws-vpc-validation_installing-aws-vpc[VPC Validation] for more information.

Reviewer: Maybe flip these around:
```suggestion
. Create a VPC with at least one private subnet. Ensure that your machine's classless inter-domain routing (CIDR) matches your virtual private cloud's CIDR. For more information, see https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-vpc.html#installation-custom-aws-vpc-requirements_installing-aws-vpc[Requirements for using your own VPC] and link:https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-vpc.html#installation-custom-aws-vpc-validation_installing-aws-vpc[VPC Validation].
```

// Module included in the following assemblies:
//
// * rosa_hcp/rosa-hcp-aws-privatelink-creating-cluster.adoc
:_mod-docs-content-type: PROCEDURE
[id="rosa-hcp-aws-privatelink-create-cluster_{context}"]
= Creating an AWS PrivateLink cluster

> You can create a private cluster with multiple availability zones (Multi-AZ) on {hcp-title} using the {product-title} (ROSA) CLI, `rosa`.

Reviewer 2: ```suggestion
You can create a private cluster with multiple availability zones (Multi-AZ) on {hcp-title} using the ROSA command line interface (CLI), `rosa`.
```

[NOTE]
====

> AWS PrivateLink is supported on existing VPCs only.

Reviewer 2: This is a little confusing to me. Do you mean "preexisting" or is there some other nuance/word we could use. It seems obvious that AWS PrivateLink would only be supported on existing VPCs; it can't be supported on non-existing VPCS because they don't exist.

====

.Prerequisites

* You have available AWS service quotas.
* You have enabled the ROSA service in the AWS Console.

> * You have installed and configured the latest {product-title} (ROSA) CLI, `rosa`, on your installation host.

Reviewer 2: ```suggestion
* You have installed and configured the latest version of the ROSA CLI on your installation host.
```

* You have created a VPC with at least one private subnet.

.Procedure

> Creating a cluster can take up to 40 minutes.

Reviewer 3: in [HCP vs Classic comparison](https://file.rdu.redhat.com/eponvell/OSDOCS-8904_Privatelink-ROSA-with-HCP/rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.html#rosa-hcp-classic-comparison_rosa-hcp-sts-creating-a-cluster-quickly), we explain that provisioning time for HCP is around 10 min.

> . You can create a {hcp-title} cluster with multiple availability zones (Multi-AZ). Your machine's classless inter-domain routing (CIDR) must match your virtual private cloud's CIDR. See https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-vpc.html#installation-custom-aws-vpc-requirements_installing-aws-vpc[Requirements for using your own VPC] and link:https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-vpc.html#installation-custom-aws-vpc-validation_installing-aws-vpc[VPC Validation] for more information.

Reviewer 4: Per the [ISG](https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#writing-steps), we should use the imperative mood in the first sentence of each step. Additionally, I don't see a clear action for a user to take in this step. 

I'd suggest either rewriting this step as an instruction to ensure the CIDR matches for this condition, or I'd move this paragraph to the prereqs, as something to ensure for people who intend to create a cluster with multiple availability zones (or whatever the condition is)

Reviewer 5: I agree

> . Create a virtual private cloud. Ensure that your machine's classless inter-domain routing (CIDR) match your virtual private cloud's CIDR. See https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-vpc.html#installation-custom-aws-vpc-requirements_installing-aws-vpc[Requirements for using your own VPC] and link:https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-vpc.html#installation-custom-aws-vpc-validation_installing-aws-vpc[VPC Validation] for more information.

Reviewer 2: ```suggestion
. Create a VPC. Ensure that your machine's classless inter-domain routing (CIDR) matches your virtual private cloud's CIDR. See https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-vpc.html#installation-custom-aws-vpc-requirements_installing-aws-vpc[Requirements for using your own VPC] and link:https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-vpc.html#installation-custom-aws-vpc-validation_installing-aws-vpc[VPC Validation] for more information.
```
This step is a little confusing given the prerequisites. The last prereq was that you have created a VPC with at least one subnet, and now the first step is to create a VPC with at least one subnet. Do they need two VPCs?

+
[IMPORTANT]
====

> If you use a firewall, you must configure it so that {product-title} can access the sites that it requires to function.

Reviewer 2: ```suggestion
If you use a firewall, you must configure it so that ROSA can access the sites required to function.
```

> For more information, see the AWS PrivateLink firewall prerequisites section.

Reviewer 4: I'd suggest putting the section title in quotes: 
```suggestion
For more information, see the "AWS PrivateLink firewall prerequisites" section.
```

====

. Create the account-wide IAM roles by running the following command:
+
[source,terminal]
----
$ rosa create account-roles --hosted-cp
----

. Create the OIDC configuration by running the following command:
+
[source,terminal]
----
$ rosa create oidc-config --mode=auto --yes
----
+

> Save the OIDC configuration ID since you need it to create the operator roles.

Reviewer 4: Per the [ISG word usage](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#s) entry on "since", it's best to only use "since" to refer to time in order to help translation:
```suggestion
Save the OIDC configuration ID because you need it to create the operator roles.
```

> Save the OIDC configuration ID since you need it to create the operator roles.

Reviewer 4: ```suggestion
Save the OIDC configuration ID since you need it to create the Operator roles.
```

+
.Example output
[source,terminal]
----
I: Setting up managed OIDC configuration
I: To create Operator Roles for this OIDC Configuration, run the following command and remember to replace <user-defined> with a prefix of your choice:
	rosa create operator-roles --prefix <user-defined> --oidc-config-id 28s4avcdt2l318r1jbk3ifmimkurk384
If you are going to create a Hosted Control Plane cluster please include '--hosted-cp'
I: Creating OIDC provider using 'arn:aws:iam::46545644412:user/user'
I: Created OIDC provider with ARN 'arn:aws:iam::46545644412:oidc-provider/oidc.op1.openshiftapps.com/28s4avcdt2l318r1jbk3ifmimkurk384'
----

. Create the Operator roles by running the following command:
+
[source,terminal]
----
$ rosa create operator-roles --hosted-cp --prefix <operator_roles_prefix> --oidc-config-id <oidc_config_id> --installer-role-arn arn:aws:iam::$<account_roles_prefix>:role/$<account_roles_prefix>-HCP-ROSA-Installer-Role
----

> . You can create a private {hcp-first} cluster by running the following command:

Reviewer 3: looks like `{hcp-first}` cannot be resolved in the documentation

> . You can create a private {hcp-title} cluster by running the following command:

Reviewer 4: Same ISG guidance about using the imperative mood:
```suggestion
. Create a private {hcp-title} cluster by running the following command:
```

If this is meant to be optional, I'd suggest writing it the same way but with "Optional:" at the beginning:
```suggestion
. Optional: Create a private {hcp-title} cluster by running the following command:
```

+
[source,terminal]
----
$ rosa create cluster --private-link --cluster-name=<cluster-name> --sts --mode=auto --hosted-cp --operator-roles-prefix <operator_role_prefix> --oidc-config-id <oidc_config_id> [--machine-cidr=<VPC CIDR>/16] --subnet-ids=<private-subnet-id1>[,<private-subnet-id2>,<private-subnet-id3>]
----

> . Enter the following command to check the status of your cluster. During cluster creation, the `State` field from the output will transition from `pending` to `installing`, and finally to `ready`.

Reviewer 2: ```suggestion
. Enter the following command to check the status of your cluster. During cluster creation, the `State` field from the output will transition from `pending` to `installing`, and finally, to `ready`.
```

+
[source,terminal]
----
$ rosa describe cluster --cluster=<cluster_name>
----
+
[NOTE]
====

> If installation fails or the `State` field does not change to `ready` after 40 minutes, check the installation troubleshooting documentation for more details.

Reviewer 3: 10 minutes instead ?

> If installation fails or the `State` field does not change to `ready` after 10 minutes, check the installation troubleshooting documentation for more details.

Reviewer 4: Suggestion to add a link to this doc somewhere in the additional resources, and to reference a specific page title if there is one.
